front/dcroom.form.php

```diff
@@ -95,6 +95,16 @@
 
 } else {
    Html::header(DCRoom::getTypeName(Session::getPluralNumber()), $_SERVER['PHP_SELF'], "management", "dcroom");
-   $room->display(['id' => $_GET["id"]]);
+   $options = [
+      'id' => $_GET["id"],
+   ];
+   if (isset($_REQUEST['_add_fromitem'])
+       && isset($_REQUEST['datacenters_id'])) {
+      $options['datacenters_id'] = $_REQUEST['datacenters_id'];
```

Comment:
I guess logic should be part of the class, not the front

Reply:
More globally, I think whe should set location from datacenter (when no location for the room itself has been set) every time (on not only when added from the datacenter button).